presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java

```diff
@@ -0,0 +1,246 @@
+                    .entity("Invalid date format").build();
+        }
+        Instant end = endDate.get();
+
+        Path filePath = LOG_DIRECTORY.resolve(filename);
+        if (!Files.isRegularFile(filePath)) {
+            return Response.status(Status.NOT_FOUND).entity("Invalid log file").build();
+        }
+
+        if (end != null) {
+            return deleteLogRange(filePath, end);
+        }
+        else {
+            try {
+                Files.write(filePath, new byte[0], StandardOpenOption.TRUNCATE_EXISTING);
+                // TODO: Use Status.ACCEPTED for asynchronous response
+                return Response.status(Status.NO_CONTENT).build();
+            }
+            catch (IOException e) {
+                // TODO: Log exception, log file disappeared since check for 404
+                return Response.status(Status.INTERNAL_SERVER_ERROR)
+                        .entity("File disappeared while processing operation").build();
+            }
+        }
+    }
+
+    private static Response deleteLogRange(Path filePath, Instant end)
+    {
+        requireNonNull(filePath);
+        requireNonNull(end);
+
+        LogFilter logFilter;
+        try {
+            logFilter = LogFilter.builder()
+                    .setFile(filePath)
+                    .setPattern(LOG_REGEX)
+                    .addGroupFilter(DATE_GROUP, getFilter(end, null))
+                    .build();
+        }
+        catch (FileNotFoundException e) {
+            // TODO: Log exception, log file disappeared since check for 404
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                    .entity("File disappeared while processing operation").build();
+        }
+
+        Stream<String> logEntries;
+        try {
+            logEntries = logFilter.streamEntries();
+        }
+        catch (IOException e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                    .entity("IOException while reading file").build();
+        }
+        catch (DateTimeParseException e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                    .entity("Attempted to read malformed date in log").build();
+        }
+
+        try {
+            Files.write(filePath, (Iterable<String>) logEntries::iterator);
+        }
+        catch (IOException e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                    .entity("IOException while writing file").build();
+        }
+
+        // TODO: Use Status.ACCEPTED for asynchronous response
+        return Response.status(Status.NO_CONTENT).build();
+    }
+
+    private static Predicate<String> getFilter(Instant start, Instant end)
+    {
+        boolean nullStart = start == null;
+        boolean nullEnd = end == null;
+        if (nullStart && nullEnd) {
+            return null;
+        }
+        else if (nullStart) {
+            return s -> DATE_FORMAT.parse(s, Instant::from).compareTo(end) <= 0;
```

Comment:
This is a simple function that consumes an argument `s` and returns the value of `DATE_FORMAT.parse(s, Instant::from).compareTo(end) <= 0`.

Because the first argument of `DateTimeFormatter::parse` is a `String`, the type of `s` is `String`.